### PR TITLE
69 - Added an endpoint to display usage data to clients

### LINKED ISSUE #69: Add an endpoint to display usage data to clients
This ticket is the first in many towards tier based services support in NC. 

The first step is to establish plumbing from InfluxDB to NC. The following flow is envisioned to achieve this goal:

1. Given a userrr, NC constructs a list of all valid api keys for that user that have been active in the past 30 days.
2. NC queries InfluxDB servers constricting the list of keys to the ones returned in step 1.
3. NC displays the returned results to front end in a format that is compatible with a charting library (TBD).

Notes:
1. NC may cache the results from Step 2 for the appropriate time window for the measurement.
2. It is assumed that we only have 1 InfluxDB instance to pull data from.
3. Apply abstraction where necessary, we may choose to opt for another database apart from InfluxDB.

### PR DESCRIPTION
**Pull request by piotrkosecki**
resolves #69 

NC gathers all currently available data from Metering API and exposes it through a single endpoint.
There was no specific information what is expected to be fetched, so when the design comes in it will probably change.

### CODE REVIEW

File: src/main/scala/tech/cryptonomic/nautilus/cloud/NautilusContext.scala [hunk 1]
```diff
@@ -86,6 +90,7 @@ trait NautilusContext extends StrictLogging {
   lazy val resourcesRepository: ResourceRepository[IO] = wire[DoobieResourceRepository[IO]]
   lazy val registrationAttemptRepository: RegistrationAttemptRepository[IO] =
     wire[InMemoryRegistrationAttemptRepository[IO]]
+  lazy val meteringApiRepository: MeteringApi[IO] = wire[MeteringApiRepository[IO]]
```
[the remark this answers is not in the document]
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/nautilus/cloud/NautilusContext.scala]: you are right, changed

File: src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/MeteringApiRepository.scala [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      .get(uri"${config.host}:${config.port}/queries/5m".params(apiKeys.map("apiKey" -> _.key): _*))
+      .readTimeout(config.readTimeout)
+      .send()
+      .map {
+        _.body.left
+          .map(QueryException(_))
+          .flatMap(
+            decode[List[ApiKeyStats]](_)
+          )
+      }
```
[the remark this answers is not in the document]
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/MeteringApiRepository.scala]: oh, my bad, I did not see that, you're completely right! changed

[the remark this answers is not in the document]
Comment (ivanopagano) [reply src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/MeteringApiRepository.scala]: The same pattern is still present in other methods below

File: src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/SttpMeteringApiRepository.scala [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+          .map(QueryException(_))
+          .flatMap(
+            decode[List[RouteStats]](_)
+          )
+      }
+  }
+
+  /** Fetches IP stats per 5m */
+  override def getIp5mStats(apiKeys: List[ApiKey]): F[Result[List[IpStats]]] = {
+    apiKeys.map("apiKey" -> _.key)
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/SttpMeteringApiRepository.scala:81]: remove

File: src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/SttpMeteringApiRepository.scala [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+          .map(QueryException(_))
+          .flatMap(
+            decode[List[IpStats]](_)
+          )
+      }
+  }
+
+  /** Fetches IP stats per 24h */
+  override def getIp24hStats(apiKeys: List[ApiKey]): F[Result[List[IpStats]]] = {
+    apiKeys.map("apiKey" -> _.key)
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/SttpMeteringApiRepository.scala:97]: remove

File: src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/SttpMeteringApiRepository.scala (hunk 3 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/SttpMeteringApiRepository.scala]: done

File: src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/SttpMeteringApiRepository.scala (hunk 4 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/nautilus/cloud/adapters/metering/api/SttpMeteringApiRepository.scala]: done

**Pull request merged**